src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -512,7 +517,20 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
           var dstConfig = InventoryUtils.constructItemRecreateConfig(checkInPiece.getReceivingTenantId(), requestContext, false);
           if (InventoryUtils.allowItemRecreate(srcConfig.tenantId(), dstConfig.tenantId())) {
             logger.info("recreateItemRecords:: recreating item by id '{}', srcTenantId: '{}', dstTenantId: '{}'", piece.getItemId(), srcConfig.tenantId(), dstConfig.tenantId());
-            futures.add(itemRecreateInventoryService.recreateItemInDestinationTenant(itemToRecreate.getCompositePoLine(), piece, srcConfig.context(), dstConfig.context()));
+
+            // Fetch purchase order asynchronously
+            Future<CompositePurchaseOrder> compositePurchaseOrderFuture = purchaseOrderStorageService
+              .getCompositeOrderByPoLineId(itemToRecreate.getPoLineId(), requestContext)
+              .recover(throwable -> {
+                logger.warn("recreateItemRecords:: Composite Purchase order can't be found for poLine: {}", itemToRecreate.getPoLineId());
+                return Future.succeededFuture(null);
```

Comment:
formatting with 2 lines should be used( these 2 lines are inside of .recover block)